### ISSUE TITLE
SIFT now working properly

### ISSUE THREAD

**Issue by mongoose54** [bytedeco/javacv#107]
I am applying SIFT feature extraction on a set of images with the following code:

*SIFT sift = new SIFT(200);
IplImage image = cvLoadImage(imageName,opencv_highgui.CV_LOAD_IMAGE_GRAYSCALE);
Mat imageMat = cvarrToMat(image);
KeyPoint keyPoints = new KeyPoint();
Mat descriptors = new Mat();
Mat mask = new Mat();
sift.detectAndCompute(imageMat,mask, keyPoints,descriptors);*

Although I explicitly set the number of key points at 200, the numbers fluctuate for each image. Is this a bug? Or something I forgot here?

**Comment by saudet** [comment]
You're not using the right constructor to set the number of keypoints. Please try again with [SIFT(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma)](http://bytedeco.org/javacpp-presets/opencv/apidocs/org/bytedeco/javacpp/opencv_nonfree.SIFT.html#SIFT-int-int-double-double-double-) and that should work.

**Comment by mongoose54** [comment]
@saudet Thanks for the reply! I changed the line per your suggestion like this: 

*SIFT sift = new SIFT(200, 3, 0.04, 10, 1.6);*

but when I print out the keypoints size I get various number of keypoints for different images:

Rows  Cols
132    128           
194    128
185    128


Only when I substituted the contrastThreshold to very small 0.0004 then I get 200.

*SIFT sift = new SIFT(200, 3, 0.0004, 10, 1.6);*

I am sharing it with you because it might help.
